### PR TITLE
Fix the types of input argument functions in the document

### LINKED ISSUE #1653: The documents of Standard Function implementations lead to misunderstanding about arguments
All children classes of Function can get not only Variable but also ndarray as their arguments. (#1221) 
However, the documents of them explain about only Variable. They lead users to misunderstanding and make #1221 meaningless.
The documents should be modified.

### PR DESCRIPTION
**Pull request by delta2323**
Fix #1653 

As I apply a simple rule to all links and functions, I need to check if the change is correct for each function and link. I also need to fix so that it obeys the 80-character rule.

### CODE REVIEW

**Comment by delta2323** [comment]
I found some Chainer functions cannot broaden the type of input arguments to `xp.ndarray`.

* loss functions in `functions.vae`: These functions impose that the inputs be `chainer.Variable` with assertion. We can safely remove these assertions and apply broaden the type.
* n_step_lstm, ctc : These functions access `data` attributes of input arrays. So they should not be the instance of `xp.ndarray`.

**Comment by takagi** [comment]
> n_step_lstm, ctc : These functions access data attributes of input arrays. So they should not be the instances of xp.ndarray.

I found that it is the same for `convolution_nd` function.

https://github.com/pfnet/chainer/blob/master/chainer/functions/connection/convolution_nd.py#L346

**Comment by unnonouno** [comment]
Functions recently updated do not support ndarrays, i think. We need to fix them.

File: chainer/functions/array/transpose_sequence.py [hunk 1]
```diff
@@ -90,7 +90,8 @@ def transpose_sequence(xs):
     :class:`~chainer.Variable`.
 
     Args:
-        xs (list of ~chainer.Variable): Variables to transpose.
+        xs (tpule of chainer.Variable, numpy.array or cupy.ndarray):
```
Comment (zori) [inline chainer/functions/array/transpose_sequence.py]: is numpy.array correct here, or should it be :class:`numpy.ndarray`?

**Comment by zori** [comment]
https://github.com/pfnet/chainer/pull/1704 allows `convolution_nd` function to take ndarray as input. The relevant documentation is in the current https://github.com/pfnet/chainer/pull/1687

File: chainer/functions/array/transpose_sequence.py (hunk 1 above)
Comment (delta2323) [inline chainer/functions/array/transpose_sequence.py]: Thank you. I fix it.

**Comment by delta2323** [comment]
> #1704 allows convolution_nd function to take ndarray as input. The relevant documentation is in the current #1687

Now, we fix the document of `convoluton_nd` as well.

**Comment by unnonouno** [comment]
We need to make tests using both Variable and ndarray.

**Comment by delta2323** [comment]
We need to solve #1797 before we merge this PR.

**Comment by delta2323** [comment]
Close this PR as it has many conflicts.

**Pull request closed without merging**